src/types.rs

```diff
@@ -651,34 +693,78 @@ impl Term {
                 .collect(),
         }
     }
-    /// The head of the Term.
+    /// Returns the head of the `Term`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, Term, parse_term, Atom};
+    ///
+    /// let mut sig = Signature::default();
+    ///
+    /// let op = sig.new_op(2, Some("S".to_string()));
+    /// let t = parse_term(&mut sig, "S(y_ z_)").expect("parse of S(y_ z_)");
```

Comment:
`S(y_ z_)` is about the right complexity for a lot of the tests in `Term`, but `S` typically has arity `0` or `3` (depending on whether its being used in an applicative TRS. Perhaps something like `A(B x_)`?